File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
import java.io.*;

class Solution {
    public List<Integer> solution(int[] answers) {
        int[] student_A = {1,2,3,4,5};
        int[] student_B = {2,1,2,3,2,4,2,5};
        int[] student_C = {3,3,1,1,2,2,4,4,5,5};
        
        List<Integer> answer = new ArrayList<>();
        int cntA = 0;
        int cntB = 0;
        int cntC = 0;
        for (int i=0; i < answers.length; i++) {
            if (answers[i] == student_A[i % student_A.length]) {
                cntA += 1;
            }
            
            if (answers[i] == student_B[i % student_B.length]) {
                cntB += 1;
            }
            
            if (answers[i] == student_C[i % student_C.length]) {
                cntC += 1;
            }
        }
        
        List<Integer> list = new ArrayList<>();
        list.add(cntA);
        list.add(cntB);
        list.add(cntC);
        
        int maxValue = Collections.max(list);
        
        if (cntA == maxValue) {
            answer.add(1);
        }
        
        if (cntB == maxValue) {
            answer.add(2);
        }
        
        if (cntC == maxValue) {
            answer.add(3);
        }
        
        Collections.sort(answer);
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    student_A = [1, 2, 3, 4, 5]
    student_B = [2, 1, 2, 3, 2, 4, 2, 5]
    student_C = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    score = [0,0,0]
    
    for i in range(len(answers)) :
        A = student_A[i % len(student_A)]
        B = student_B[i % len(student_B)]
        C = student_C[i % len(student_C)]
        
        if answers[i] == A :
            score[0] += 1
        if answers[i] == B :
            score[1] += 1
        if answers[i] == C :
            score[2] += 1
        
    maxValue = max(score)
    for i in range(len(score)) :
        if maxValue == score[i] :
            answer.append(i+1)
    return answer